File: 프로그래머스/lv3/12938. 최고의 집합/최고의 집합.py
def solution(n, s):
    if n > s:
        return [-1]
    
    # s를 n으로 나눈 값을 n번 반복, 남은 mod를 answer에 1씩 더해줌
    div, mod = divmod(s, n)
    answer = [div] * n
    
    while mod > 0:
        answer[n - 1] += 1
        n -= 1
        mod -= 1

    return answer

File: 프로그래머스/3/12938. 최고의 집합/최고의 집합.java
import java.util.Arrays;

class Solution {

    public int[] solution(int n, int s) {
        if (s < n) {
            return new int[]{-1};
        }

        int quotient = s / n;
        int remainder = s % n;

        int[] answer = new int[n];
        Arrays.fill(answer, quotient);

        for (int i = n - remainder; i < n; i++) {
            answer[i]++;
        }

        return answer;
    }
}